### PR TITLE
Add documentation demonstrating support of enum of objects

### PR DESCRIPTION
**Pull request by robnewman**
# Description
Add docs illustrating support of compound enums, with example object keyword

### CODE REVIEW

File: docs/advanced-features/workflows.md [hunk 1]
```diff
@@ -354,6 +354,78 @@ With this Schema lists such as
 `["Fixed 1", "Any string", 123]` 
 are valid but `["Any string", 123]` are invalid.
 
+### Objects
+Quilt also supports the [`object` data type](https://json-schema.org/understanding-json-schema/reference/object.html).
```
Comment (akarve) [inline docs/advanced-features/workflows.md:358]: This section is confusing. Eliminate or rework opening. It's not about Quilt but about the validators we support. I'd nest these under `### Example properties` and say "The following example show how you can specify complex properties such as objects and arrays with arbitrary nesting." And then `####` h4 for subsequent sections.

File: docs/advanced-features/workflows.md [hunk 2]
```diff
@@ -354,6 +354,78 @@ With this Schema lists such as
...
+            "name": {
+              "default": "Optional default value",
+              "type": "string"
+            }
+        }
+    }
+}
+```
+
+### Compound enums
```
Comment (akarve) [inline docs/advanced-features/workflows.md:382]:
```suggestion
### Enums of arrays and objects
```

**Review by akarve — CHANGES_REQUESTED** [review]

File: docs/advanced-features/workflows.md (hunk 1 above)
Comment (robnewman) [reply docs/advanced-features/workflows.md]: Fixed

File: docs/advanced-features/workflows.md (hunk 2 above)
Comment (robnewman) [reply docs/advanced-features/workflows.md]: Added explicit h4 titles for each example code snippets

File: docs/advanced-features/workflows.md [hunk 3]
```diff
@@ -354,6 +354,59 @@ With this Schema lists such as
 `["Fixed 1", "Any string", 123]` 
 are valid but `["Any string", 123]` are invalid.
 
+### Example properties
+The following examples show how you can specify complex properties
+such as `object` and `array` data types with arbitrary nesting.
+
```
Comment (akarve) [inline docs/advanced-features/workflows.md:360]: Looks good please add back `#### Objects section` no explanation and we're good.

File: docs/advanced-features/workflows.md [hunk 4]
```diff
@@ -354,6 +354,59 @@ With this Schema lists such as
 `["Fixed 1", "Any string", 123]` 
 are valid but `["Any string", 123]` are invalid.
 
+### Example properties
+The following examples show how you can specify complex properties
```
Comment (akarve) [inline docs/advanced-features/workflows.md:358]:
```suggestion
The following examples show how you can specify complex `properties`
```

File: docs/advanced-features/workflows.md [hunk 5]
```diff
@@ -354,6 +354,59 @@ With this Schema lists such as
...
+such as `object` and `array` data types with arbitrary nesting.
```
Comment (akarve) [inline docs/advanced-features/workflows.md:359]:
```suggestion
such as `object` and `array` data types with arbitrary nesting and `enum` types.
```

**Review by akarve — CHANGES_REQUESTED** [review]

File: docs/advanced-features/workflows.md (hunk 3 above)
Comment (robnewman) [reply docs/advanced-features/workflows.md:360]: `#### Objects` section re-added (code snippet) and no explanation

File: docs/advanced-features/workflows.md (hunk 4 above)
Comment (robnewman) [reply docs/advanced-features/workflows.md:358]: Inline markup added

File: docs/advanced-features/workflows.md (hunk 5 above)
Comment (robnewman) [reply docs/advanced-features/workflows.md:359]: Fixed

**Review by akarve — APPROVED** [review]

**Pull request merged**